control/plugin_manager.go

```diff
@@ -485,7 +485,8 @@ func (p *pluginManager) UnloadPlugin(pl core.Plugin) (*loadedPlugin, serror.Snap
 	}
 
 	// If the plugin was loaded from os.TempDir() clean up
-	if strings.Contains(plugin.Details.Path, os.TempDir()) {
+	subdir := strings.Replace(plugin.Details.ExecPath, os.TempDir()+"/", "", 1)
```

Comment:
I would like to add a more complete comment explaining what is happening here. When looking at this issue it took me a second to know why we were removing plugins loaded from TempDir. Not just the what but the why so that the next time we look at this we know that we are trying to remove plugins loaded via rest (b/c they get written to os.TempDir()) but want to avoid deleting plugins loaded via auto-discover that might also reside in the os.TempDir().